kiwi_lint/generic_foreign_key.py

```diff
@@ -0,0 +1,34 @@
+import astroid
+from pylint.checkers import utils, BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint_django.utils import node_is_subclass
+
+
+class GenericForeignKeyChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+
+    name = 'generic-foreign-key-checker'
+
+    msgs = {
+        'R4525': (
+            'Avoid using GenericForeignKey',
+            'avoid-generic-foreign-key',
+            'Avoid using django.contrib.contenttypes.fields.GenericForeignKey',
```

Comment:
long help text needs a better explanation why this field could be a problem. Given this mostly comes from 3rd party apps we can't really change it but we can make sure we are handling deletions and/or other problematic areas gracefully (which by itself could be another linter or an extension to this one).